script.sublissimo/resources/lib/script.py

```diff
@@ -0,0 +1,452 @@
+    if choice == -1 or choice == 3:
+        show_dialog(subtitlefile, filename)
+    if choice == 4:
+        sys.exit()
+    if choice == 0:
+        new_file_name = filename[:-4] + "_edited.srt"
+    if choice == 1:
+        new_file_name = filename
+    if choice == 2:
+        # Give new filename
+        new_file_name = xbmcgui.Dialog().input(_(32042), defaultt=filename)
+    with closing(File(new_file_name, 'w')) as fo:
+        fo.write("".join(subtitlefile))
+    backupfile = copy.deepcopy(subtitlefile)
+    if xbmcvfs.exists(new_file_name):
+        # written to, to use select in kodi sub menu
+        xbmcgui.Dialog().ok(_(32043), new_file_name + _(32044))
+    else:
+        #Error, File not written
+        xbmcgui.Dialog().ok(_(32014), _(32045))
+    show_dialog(subtitlefile, filename)
+
+def exiting(subtitlefile=[], filename=""):
+    global backupfile
+    if backupfile != subtitlefile:
+        # Warning, You might have unsaved progress, Exit anyway, Save
+        ret = xbmcgui.Dialog().yesno(_(32046), _(32047),
+                                      nolabel=_(32048), yeslabel=_(32049))
+        if ret:
+            save_the_file(subtitlefile, filename)
+    sys.exit()
+
+def stretch_subtitle(subtitlefile, filename):
+    #Write new timestamp, for example
+    xbmcgui.Dialog().ok(_(31001), _(32028))
+    timestring = xbmcgui.Dialog().input(_(32029))
+    try:
+        current_sub = Subtitle(subtitlefile)
+        subtitlefile = current_sub.create_new_factor(timestring)
+    except:
+        xbmcgui.Dialog().ok(_(31001), _(32028))
+        show_dialog(subtitlefile, filename)
+    # All subtitles
+    xbmcgui.Dialog().multiselect(_(32010), subtitlefile)
+    show_dialog(subtitlefile, filename)
+
+def make_timelines_classical(decimal):
+    hours = int(decimal / 3600000)
+    restminutes = decimal % 3600000
+    minutes = int(restminutes / 60000)
+    restseconds = restminutes % 60000
+    seconds = int(restseconds / 1000)
+    milliseconds = int(restseconds % 1000)
+    output = (str(hours).zfill(2) + ":" + str(minutes).zfill(2) + ":" +
+              str(seconds).zfill(2) + "," + str(milliseconds).zfill(3))
+    return output
+
+def tryout(starting_time, ending_time, subtitlefile, filename):
+    start = make_timelines_classical(starting_time*1000)
+    end = make_timelines_classical(ending_time*1000)
+    current_start_sub = Subtitle(subtitlefile)
+    subtitlefile2 = current_start_sub.move_subtitles(start)
+    # line_index = xbmcgui.Dialog().select("Result", subtitlefile2)
+    current_sub = Subtitle(subtitlefile2)
+    subtitlefile3 = current_sub.create_new_factor(end)
+    # xbmcgui.Dialog().multiselect("All subtitles", subtitlefile3)
+    if subtitlefile3:
+        # Succes, Your subs starts at, your subs end at.
+        xbmcgui.Dialog().ok(_(32017), _(32036) + start + "\n" +
+                                      _(32037) + end)
+    show_dialog(subtitlefile3, filename)
+
+def retrieve_video(subtitlefile, filename):
```

Comment:
Sorry, you have to change this. This is too error prone. You need to get the playing file from the Player object itself and compare it to your subtitle (carefull with xbmcvfs locations / http streams / etc). The location that would then be passed to the player would be the unique path associated with the current playing item (in the form of videodb://xxx) otherwise a new entry is added to the kodi database all the time.

Reply:
Ok. So (I hope i'm interpreting this right) if I instead let the user select a video file directly from the videodb something like: 

location = xbmcgui.Dialog().browse(1, _(32020), 'video', '', False, False, "videodb://")

the location passed to the player would indeed be a "videodb://xxx" unique path. 
Would that be a possible solution?

Reply:
Yeah sure, as long as the path is a unique xbmc VFS file (videodb://, plugin://, Pvr://) it is fine. We are not trying to block player.play() just using it right